integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/highlight/HighlighterFastVectorIT.java

```diff
@@ -0,0 +1,222 @@
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "another" ) )
+				.highlighter( h2 -> h2.highlighterFastVector().tagSchema( HighlighterTagSchema.STYLED ) )
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						Arrays.asList( "some <em class=\"hlt1\">another</em> value" ),
+						Arrays.asList( "some yet <em class=\"hlt1\">another</em> value" )
+				);
+	}
+
+	@Test
+	public void boundaryCharacters() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "time" ) )
+				.highlighter( h2 -> h2.highlighterFastVector()
+						.boundaryScanner()
+						.chars()
+						.locale( Locale.ENGLISH )
+						.boundaryChars( "-" )
+						.boundaryMaxScan( 5 )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						// some unexplainable result >_<
+						Arrays.asList( "at has - dash in - it from <em>time</em> - to -some useless text in between - <em>time</em> to see - how - boundary_chars " )
+				);
+	}
+
+	@Test
+	public void phraseLimit() {
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "rock" ) )
+				.highlighter( h2 -> h2.highlighterFastVector()
+						.phraseLimit( 1 )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						Arrays.asList( "Scorpions are a German <em>rock</em> band formed in Hanover in 1965 by guitarist Rudolf Schenker. Since the band's" )
+				);
+	}
+
+	@Test
+	public void fragmentOffset() {
+		assumeTrue( "Ignoring the test on purpose as the results look not ok.", false );
+		StubMappingScope scope = index.createScope();
+
+		SearchQuery<List<String>> highlights = scope.query().select(
+						f -> f.highlight( "string" ).highlighter( "offset" )
+				)
+				.where( f -> f.match().field( "string" ).matching( "rock" ) )
+				// `fragmentOffset` doesn't work on a global level O_O
+				.highlighter( "offset", h2 -> h2.highlighterFastVector()
+						// should skip first sentence
+						.boundaryScanner()
+							.sentence().locale( Locale.ENGLISH )
+						.fragmentOffset( 180 )
+						.fragmentSize( 1000 )
+				)
+				.toQuery();
+
+		assertThatHits( highlights.fetchAllHits() )
+				.hasHitsAnyOrder(
+						// returns just all text highlighted... expected that it would skip some initial part
+						// >> *fragment_offset* controls the margin from which you want to start highlighting. Only valid when using the fvh highlighter
+						Arrays.asList( "Since the band's inception, its musical style has ranged from hard <em>rock</em>, heavy metal and glam metal to soft <em>rock</em>." )
```

Comment:
The question is: what's the unit of `fragmentOffset`; is it a number of characters, of fragments, or (considering we're using the boundary scanner) of sentences, ... ?

Reply:
ok... so my initial interpretation was wrong. it's not about skipping the text but rather about including additional text at the start of the fragment. The way this property is supposed to work is - "how much text should it also capture before the highlighted element". Let's say there's the text:
```
Since the band's inception. Its musical style has ranged from hard rock. Heavy metal and glam metal. To soft rock.
```
and we are searching for `ranged` then if we'd want to also include the "first sentence" - `Since the band's inception` we'd say this margin to a value that is more than ` " Its musical style has ".lenght()` so that with this margin applied we are also capturing the previous sentence. So a loose translation of this property would be - how many characters to look back from a matched ("highlighted") token to find a "boundary" at which to start a fragment that will be returned to the user.